src/hotspot/share/gc/shenandoah/shenandoahControlThread.cpp

```diff
@@ -308,7 +312,7 @@ void ShenandoahControlThread::run_service() {
       // Clear metaspace oom flag, if current cycle unloaded classes
       if (heap->unload_classes()) {
         // HEY! Should we do this if the cycle was cancelled/degenerated?
-        assert(generation != YOUNG, "should not unload classes in young cycle");
+        assert(generation == GLOBAL, "Only unload classes during GLOBAL cycle");
```

Comment:
Is the comment still valid? Also, maybe change 'HEY!' to TODO so that IDE would flag it.